### DISCUSSION TITLE
How to get a URL and title server-side?

### DISCUSSION THREAD

**Discussion by adaboese** [remix-run/remix#8953] [General]
My use case is that I am moving Google Analytics from client-side to server-side.

I need to send two parameters to Google Measurement API: current URL and the page title.

I know how to get the current URL inside of a loader. However, there is no way to get the current page title inside of the loader.

I am not sure where would I need to add my code to get URL and title for every request.

Is there some hook I could fire after the page has been rendered? That would give me the title too

**Comment by huijiewei** [comment]
you should define title in loader, meta and Component get title from loader

**Reply by adaboese** [reply]
The top most loader is not going to have information about the title as it may be overridden by a descending loader.

And I don't want to copy analytics code to every every page as that would be very error prone.

**Reply by huijiewei** [reply]
The top loader only runs when the user enters the website or when the browser refreshes the page.

**Reply by adaboese** [reply]
Yes, but it does not have access to the page title.

**Comment by sergiodxa** [comment]
There's no way, because loaders run before MetaFunction and before the rendering

The only place you could do it is inside entry.server where you have access to EntryContext which contains the data from the loaders and meta, but then it will only run on SSR.

**Reply by adaboese** [reply]
That seems fine for the analytics use case.

Can you elaborate a bit more on how would I get a reference to the title from entry.server?

**Reply by sergiodxa** [reply]
For example doing `remixContext.routeModules["root"].meta` will give you access to root meta, you can iterate `remixContext.routeModules` to get the meta of every route, combine with with `remixContext.staticHandlerContext.loaderData` to get the loader data and you can call MetaFunction there to get the final title
